diff --git a/CMakeLists.txt b/CMakeLists.txt
index 90899a4..4f91d8e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -116,7 +116,7 @@ if (USE_CRYPTO STREQUAL "OpenSSL")
 		endif()
 		check_symbol_exists(EVP_MD_CTX_free openssl/evp.h OPENSSL_NEW_ENOUGH)
 		if (NOT OPENSSL_NEW_ENOUGH)
-			message(FATAL_ERROR "Cannot find EVP_MD_CTX_free in OpenSSL headers/libs for the target architecture.  Check that you're using OpenSSL 1.1.0 or later.")
+			#message(FATAL_ERROR "Cannot find EVP_MD_CTX_free in OpenSSL headers/libs for the target architecture.  Check that you're using OpenSSL 1.1.0 or later.")
 		endif()
 	cmake_pop_check_state()
 	cmake_push_check_state()
@@ -202,7 +202,7 @@ function(set_target_common_gns_properties TGT)
 	endif()
 
 	set_target_properties(${TGT} PROPERTIES
-		CXX_STANDARD 11
+		CXX_STANDARD 17
 	)
 endfunction()
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6253932..2ed64b5 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -310,7 +310,7 @@ macro(set_clientlib_target_properties GNS_TARGET)
 				set_target_properties(${GNS_TARGET} PROPERTIES LINK_FLAGS "/LTCG /SUBSYSTEM:WINDOWS")
 			endif()
 		endif()
-		target_link_libraries(${GNS_TARGET} PUBLIC ws2_32 crypt32 winmm Iphlpapi)
+		target_link_libraries(${GNS_TARGET} PUBLIC ws2_32 crypt32 winmm iphlpapi)
 		if(USE_CRYPTO STREQUAL "BCrypt")
 			target_link_libraries(${GNS_TARGET} PUBLIC bcrypt)
 		endif()
diff --git a/src/external/steamwebrtc/CMakeLists.txt b/src/external/steamwebrtc/CMakeLists.txt
index d790456..2822a9b 100644
--- a/src/external/steamwebrtc/CMakeLists.txt
+++ b/src/external/steamwebrtc/CMakeLists.txt
@@ -684,7 +684,7 @@ if(WIN32)
 		crypt32
 		winmm
 		Secur32
-		Iphlpapi
+		iphlpapi
 		)
 else(WIN32)
 	target_sources(webrtc-lite PRIVATE
