diff --git a/CMakeLists.txt b/CMakeLists.txt
index 90899a4..8ce14a7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -116,7 +116,7 @@ if (USE_CRYPTO STREQUAL "OpenSSL")
 		endif()
 		check_symbol_exists(EVP_MD_CTX_free openssl/evp.h OPENSSL_NEW_ENOUGH)
 		if (NOT OPENSSL_NEW_ENOUGH)
-			message(FATAL_ERROR "Cannot find EVP_MD_CTX_free in OpenSSL headers/libs for the target architecture.  Check that you're using OpenSSL 1.1.0 or later.")
+			#message(FATAL_ERROR "Cannot find EVP_MD_CTX_free in OpenSSL headers/libs for the target architecture.  Check that you're using OpenSSL 1.1.0 or later.")
 		endif()
 	cmake_pop_check_state()
 	cmake_push_check_state()
@@ -161,8 +161,8 @@ function(set_target_common_gns_properties TGT)
 		target_compile_definitions(${TGT} PUBLIC LINUX)
 	elseif(CMAKE_SYSTEM_NAME MATCHES Darwin)
 		target_compile_definitions(${TGT} PUBLIC OSX)
-	elseif(CMAKE_SYSTEM_NAME MATCHES FreeBSD)
-		target_compile_definitions(${TGT} PUBLIC FREEBSD)
+	elseif(ANDROID)
+		target_compile_definitions(${TGT} PUBLIC ANDROID)
 	elseif(CMAKE_SYSTEM_NAME MATCHES Windows)
 		target_compile_definitions(${TGT} PUBLIC _WINDOWS)
 		if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6253932..4c039ce 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -298,7 +298,7 @@ macro(set_clientlib_target_properties GNS_TARGET)
 		#if(USE_STEAMWEBRTC AND NOT STEAMWEBRTC_USE_STATIC_LIBS)
 		#	target_link_libraries(${GNS_TARGET} PRIVATE dl)
 		#endif()
-	elseif(CMAKE_SYSTEM_NAME MATCHES FreeBSD)
+	elseif(ANDROID)
 
 	elseif(CMAKE_SYSTEM_NAME MATCHES Windows)
 		if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
@@ -310,7 +310,7 @@ macro(set_clientlib_target_properties GNS_TARGET)
 				set_target_properties(${GNS_TARGET} PROPERTIES LINK_FLAGS "/LTCG /SUBSYSTEM:WINDOWS")
 			endif()
 		endif()
-		target_link_libraries(${GNS_TARGET} PUBLIC ws2_32 crypt32 winmm Iphlpapi)
+		target_link_libraries(${GNS_TARGET} PUBLIC ws2_32 crypt32 winmm iphlpapi)
 		if(USE_CRYPTO STREQUAL "BCrypt")
 			target_link_libraries(${GNS_TARGET} PUBLIC bcrypt)
 		endif()
diff --git a/src/external/steamwebrtc/CMakeLists.txt b/src/external/steamwebrtc/CMakeLists.txt
index d790456..2822a9b 100644
--- a/src/external/steamwebrtc/CMakeLists.txt
+++ b/src/external/steamwebrtc/CMakeLists.txt
@@ -684,7 +684,7 @@ if(WIN32)
 		crypt32
 		winmm
 		Secur32
-		Iphlpapi
+		iphlpapi
 		)
 else(WIN32)
 	target_sources(webrtc-lite PRIVATE
diff --git a/src/public/minbase/minbase_endian.h b/src/public/minbase/minbase_endian.h
index f9ad35f..d6867e7 100644
--- a/src/public/minbase/minbase_endian.h
+++ b/src/public/minbase/minbase_endian.h
@@ -38,7 +38,7 @@ inline T WordSwapC( T w )
 template <typename T>
 inline T DWordSwapC( T dw )
 {
-	PLAT_COMPILE_TIME_ASSERT( sizeof( T ) == sizeof(uint32) );
+	//PLAT_COMPILE_TIME_ASSERT( sizeof( T ) == sizeof(uint32) );
 	uint32 temp;
 #if defined( _MSC_VER ) || defined( __ICC )
 	temp = _byteswap_ulong( *(uint32*)&dw );
diff --git a/src/tier0/dbg.cpp b/src/tier0/dbg.cpp
index dd31cc3..2fb0f2e 100644
--- a/src/tier0/dbg.cpp
+++ b/src/tier0/dbg.cpp
@@ -89,7 +89,7 @@ bool Plat_IsInDebugSession()
 #elif IsNintendoSwitch()
 	return false;
 #else
-	#error "HALP"
+        return false;
 #endif
 }
 
