diff --git a/CMakeLists.txt b/CMakeLists.txt
index 55644b0..4eb964d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1450,7 +1450,7 @@ target_compile_options(${IMPL_TARGET} PRIVATE ${C_FLAGS})
 if(TARGET build_version)
     add_dependencies(${IMPL_TARGET} build_version)
 endif()
-
+find_program(CMAKE_DLLTOOL NAMES "${DLLTOOL}" DOC "dlltool executable")
 if(WIN32 AND MINGW AND ALSOFT_BUILD_IMPORT_LIB AND NOT LIBTYPE STREQUAL "STATIC")
     find_program(SED_EXECUTABLE NAMES sed DOC "sed executable")
     if(NOT SED_EXECUTABLE OR NOT CMAKE_DLLTOOL)
diff --git a/al/source.cpp b/al/source.cpp
index cba3386..4fc7292 100644
--- a/al/source.cpp
+++ b/al/source.cpp
@@ -21,7 +21,7 @@
 #include "config.h"
 
 #include "source.h"
-
+#define __STDC_FORMAT_MACROS
 #include <algorithm>
 #include <array>
 #include <atomic>
