diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9686ecb..49a4e5f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -532,7 +532,7 @@ if(USE_GCC OR USE_CLANG)
     if(HAVE_GCC_WERROR_DECLARATION_AFTER_STATEMENT)
-      list(APPEND EXTRA_CFLAGS "-Werror=declaration-after-statement")
+      #list(APPEND EXTRA_CFLAGS "-Werror=declaration-after-statement")
     endif()
     list(APPEND EXTRA_CFLAGS "-Wdeclaration-after-statement")
   endif()

   if(DEPENDENCY_TRACKING)
