diff --git a/CMakeLists.txt b/CMakeLists.txt
index aaf3bf1..21e1605 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,9 +113,9 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CLANG)
 elseif(MSVC)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -utf-8")
   # Without /Zc:__cplusplus, MSVC does not define the right value for
-  # __cplusplus. See https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/
-  # If this becomes too problematic for downstream code, we can look at
-  # _MSVC_LANG.
+if (${CMAKE_GENERATOR} MATCHES "^Visual")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
+endif ()
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -utf-8 -Zc:__cplusplus")
 endif()
 
