diff --git a/crypto/CMakeLists.txt b/crypto/CMakeLists.txt
index a364e86..32596ac 100644
--- a/crypto/CMakeLists.txt
+++ b/crypto/CMakeLists.txt
@@ -1058,7 +1058,7 @@ add_library(crypto $<TARGET_OBJECTS:crypto_obj> empty.c)
 export_symbol(crypto ${CMAKE_CURRENT_BINARY_DIR}/crypto_p.sym)
 target_link_libraries(crypto ${PLATFORM_LIBS})
 if (WIN32)
-	set(CRYPTO_POSTFIX -${CRYPTO_MAJOR_VERSION})
+	#set(CRYPTO_POSTFIX -${CRYPTO_MAJOR_VERSION})
 endif()
 set_target_properties(crypto PROPERTIES
 	OUTPUT_NAME crypto${CRYPTO_POSTFIX}
diff --git a/ssl/CMakeLists.txt b/ssl/CMakeLists.txt
index f5d52f7..7f9a08e 100644
--- a/ssl/CMakeLists.txt
+++ b/ssl/CMakeLists.txt
@@ -86,7 +86,7 @@ endif()
 export_symbol(ssl ${CMAKE_CURRENT_SOURCE_DIR}/ssl.sym)
 target_link_libraries(ssl crypto ${PLATFORM_LIBS})
 if (WIN32)
-	set(SSL_POSTFIX -${SSL_MAJOR_VERSION})
+	#set(SSL_POSTFIX -${SSL_MAJOR_VERSION})
 endif()
 set_target_properties(ssl PROPERTIES
 	OUTPUT_NAME ssl${SSL_POSTFIX}
diff --git a/tls/CMakeLists.txt b/tls/CMakeLists.txt
index 62dde63..eb19528 100644
--- a/tls/CMakeLists.txt
+++ b/tls/CMakeLists.txt
@@ -53,7 +53,7 @@ add_library(tls $<TARGET_OBJECTS:tls_obj> $<TARGET_OBJECTS:ssl_obj>
 export_symbol(tls ${CMAKE_CURRENT_BINARY_DIR}/tls.sym)
 target_link_libraries(tls ${PLATFORM_LIBS})
 if (WIN32)
-	set(TLS_POSTFIX -${TLS_MAJOR_VERSION})
+	#set(TLS_POSTFIX -${TLS_MAJOR_VERSION})
 endif()
 set_target_properties(tls PROPERTIES
 	OUTPUT_NAME tls${TLS_POSTFIX}
