diff --git a/CMakeLists.txt b/CMakeLists.txt
index b85a7e8..1e7928e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 # CMake build file list for OpenAL
 
-cmake_minimum_required(VERSION 3.13)
+cmake_minimum_required(VERSION 3.15)
 enable_testing()
 
 if(APPLE)
@@ -282,12 +282,12 @@ else()
         set(C_FLAGS ${C_FLAGS} $<$<COMPILE_LANGUAGE:CXX>:-Wno-interference-size>)
     endif()
 
-    if(ALSOFT_WERROR)
-        set(C_FLAGS ${C_FLAGS} -Werror)
-    else()
-        set(C_FLAGS ${C_FLAGS} -Werror=undef)
-    endif()
-
+    if (LIBTYPE STREQUAL "SHARED" AND NOT MSVC AND NOT APPLE)
+        string(APPEND CMAKE_CXX_FLAGS " -g0 -s")
+    endif ()
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND MINGW)
+        string(APPEND CMAKE_CXX_FLAGS " -femulated-tls")
+    endif ()
     # NOTE: This essentially provides the equivalent of the C++26 feature to
     # initialize all local variables with a non-0 bit pattern. Until C++26 is
     # adopted, the [[gnu::uninitialized]] attribute will avoid the auto-
@@ -1555,7 +1555,7 @@ target_compile_options(${IMPL_TARGET} PRIVATE ${C_FLAGS})
 if(TARGET alsoft.build_version)
     add_dependencies(${IMPL_TARGET} alsoft.build_version)
 endif()
-
+find_program(CMAKE_DLLTOOL NAMES "${DLLTOOL}" DOC "dlltool executable")
 if(WIN32 AND MINGW AND ALSOFT_BUILD_IMPORT_LIB AND NOT LIBTYPE STREQUAL "STATIC")
     find_program(SED_EXECUTABLE NAMES sed DOC "sed executable")
     if(NOT SED_EXECUTABLE OR NOT CMAKE_DLLTOOL)
diff --git a/al/source.cpp b/al/source.cpp
index f045d1c..ae2a9ed 100644
--- a/al/source.cpp
+++ b/al/source.cpp
@@ -21,7 +21,7 @@
 #include "config.h"
 
 #include "source.h"
-
+#define __STDC_FORMAT_MACROS
 #include <algorithm>
 #include <array>
 #include <atomic>
