diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0cf0613..ee457d5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1324,28 +1324,28 @@ if(TARGET build_version)
     add_dependencies(${IMPL_TARGET} build_version)
 endif()
 
-if(WIN32 AND MINGW AND ALSOFT_BUILD_IMPORT_LIB AND NOT LIBTYPE STREQUAL "STATIC")
-    find_program(SED_EXECUTABLE NAMES sed DOC "sed executable")
-    if(NOT SED_EXECUTABLE OR NOT CMAKE_DLLTOOL)
-        message(STATUS "")
-        if(NOT SED_EXECUTABLE)
-            message(STATUS "WARNING: Cannot find sed, disabling .def/.lib generation")
-        endif()
-        if(NOT CMAKE_DLLTOOL)
-            message(STATUS "WARNING: Cannot find dlltool, disabling .def/.lib generation")
-        endif()
-    else()
-        set_property(TARGET OpenAL APPEND_STRING PROPERTY
+IF(WIN32 AND MINGW AND ALSOFT_BUILD_IMPORT_LIB AND NOT LIBTYPE STREQUAL "STATIC")
+    FIND_PROGRAM(SED_EXECUTABLE NAMES sed DOC "sed executable")
+    FIND_PROGRAM(DLLTOOL_EXECUTABLE NAMES "${DLLTOOL}" DOC "dlltool executable")
+    IF(NOT SED_EXECUTABLE OR NOT DLLTOOL_EXECUTABLE)
+        MESSAGE(STATUS "")
+        IF(NOT SED_EXECUTABLE)
+            MESSAGE(STATUS "WARNING: Cannot find sed, disabling .def/.lib generation")
+        ENDIF()
+        IF(NOT DLLTOOL_EXECUTABLE)
+            MESSAGE(STATUS "WARNING: Cannot find dlltool, disabling .def/.lib generation")
+        ENDIF()
+    ELSE()
+        SET_PROPERTY(TARGET OpenAL APPEND_STRING PROPERTY
             LINK_FLAGS " -Wl,--output-def,OpenAL32.def")
-        add_custom_command(TARGET OpenAL POST_BUILD
+        ADD_CUSTOM_COMMAND(TARGET OpenAL POST_BUILD
             COMMAND "${SED_EXECUTABLE}" -i -e "s/ @[^ ]*//" OpenAL32.def
-            COMMAND "${CMAKE_DLLTOOL}" -d OpenAL32.def -l OpenAL32.lib -D OpenAL32.dll
+            COMMAND "${DLLTOOL_EXECUTABLE}" -d OpenAL32.def -l OpenAL32.lib -D OpenAL32.dll
             COMMENT "Stripping ordinals from OpenAL32.def and generating OpenAL32.lib..."
             VERBATIM
         )
-    endif()
-endif()
-
+    ENDIF()
+ENDIF()
 if(HAS_ROUTER)
     message(STATUS "")
     message(STATUS "Building DLL router")
