diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ecfe1e..a04e239 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,7 @@ include(cmake/ConfigureOpenSSL.cmake)
 include(cmake/DetectTargetPlatform.cmake)
 include(cmake/FetchOpenSSL.cmake)
 include(cmake/FindVcvarsall.cmake)
-include(cmake/GetCPM.cmake)
+#include(cmake/GetCPM.cmake)
 
 # Custom options
 option(OPENSSL_BUILD_VERBOSE "Enable verbose output from build" OFF)
@@ -31,22 +31,22 @@ option(OPENSSL_INSTALL "Install OpenSSL components to the <prefix> directory" OF
 option(OPENSSL_INSTALL_CERT "Install cert.pem to the <openssldir> directory" OFF)
 option(OPENSSL_TEST "Enable testing and build OpenSSL self tests" OFF)
 option(OPENSSL_USE_CCACHE "Use ccache if available" ON)
-
+option(OPENSSL_NO_ASM "Disable asm" ON)
 if("${OPENSSL_BUILD_TARGET}" STREQUAL "")
-    # Makefile target for build
     set(OPENSSL_BUILD_TARGET build_libs)
 endif()
-
 if("${OPENSSL_INSTALL_TARGET}" STREQUAL "")
-    # Makefile target for install
     set(OPENSSL_INSTALL_TARGET install_dev)
 endif()
-
 if("${OPENSSL_TARGET_PLATFORM}" STREQUAL "")
-    # Use OpenSSL's Configure target
     detect_target_platform(OPENSSL_TARGET_PLATFORM)
 endif()
-
+if (OPENSSL_NO_ASM)
+    list(APPEND OPENSSL_CONFIGURE_OPTIONS no-asm)
+endif()
+if (NOT EXISTS ${CMAKE_SOURCE_DIR}/openssl)
+    execute_process(WORKING_DIRECTORY ${CMAKE_SOURCE_DIR} COMMAND git clone -b openssl-${OPENSSL_TARGET_VERSION} --depth 1 https://github.com/openssl/openssl.git)
+endif ()
 # Fetch OpenSSL source
 fetch_openssl()
 
