diff --git a/CMakeLists.txt b/CMakeLists.txt
index 120dd38..95e2bfe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -234 +234 @@ if (sol2-is-top-level-project)
-			find_package(Lua 5.1 EXACT REQUIRED)
+			find_package(Lua 5.1 EXACT)
@@ -237 +237 @@ if (sol2-is-top-level-project)
-			find_package(Lua 5.2 EXACT REQUIRED)
+			find_package(Lua 5.2 EXACT)
@@ -240 +240 @@ if (sol2-is-top-level-project)
-			find_package(Lua 5.3 EXACT REQUIRED)
+			find_package(Lua 5.3 EXACT)
@@ -243 +243 @@ if (sol2-is-top-level-project)
-			find_package(Lua 5.4 EXACT REQUIRED)
+			find_package(Lua 5.4 EXACT)
