From cd6a121a268c09d3e21be428255e0d2f8a791809 Mon Sep 17 00:00:00 2001
From: Andrey Vasiliev <foursilico@gmail.com>
Date: Sat, 21 Nov 2020 21:08:44 +0200
Subject: [PATCH] Skip shadow culling

---
 OgreMain/src/OgreSceneManager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/OgreMain/src/OgreSceneManager.cpp b/OgreMain/src/OgreSceneManager.cpp
index e731a37..928deec 100644
--- a/OgreMain/src/OgreSceneManager.cpp
+++ b/OgreMain/src/OgreSceneManager.cpp
@@ -364,7 +364,7 @@ void SceneManager::_populateLightList(const Vector3& position, Real radius,
         // Calc squared distance
         lt->_calcTempSquareDist(position);

-        if (lt->getType() == Light::LT_DIRECTIONAL)
+        if (lt->getType() == Light::LT_DIRECTIONAL || (lt->getCastShadows() && isShadowTechniqueIntegrated()))
         {
             // Always included
             destList.push_back(lt);
--
2.25.1

