diff --git a/RenderSystems/GLSupport/src/EGL/OgreEGLSupport.cpp b/RenderSystems/GLSupport/src/EGL/OgreEGLSupport.cpp
index bfe55652..f3462de0 100755
--- a/RenderSystems/GLSupport/src/EGL/OgreEGLSupport.cpp
+++ b/RenderSystems/GLSupport/src/EGL/OgreEGLSupport.cpp
@@ -386,8 +386,8 @@ namespace Ogre {
         StringStream ext;
 
         // client extensions
-        propStr = eglQueryString(EGL_NO_DISPLAY, EGL_EXTENSIONS);
-        ext << propStr << " ";
+        // propStr = eglQueryString(EGL_NO_DISPLAY, EGL_EXTENSIONS);
+        // ext << propStr << " ";
 
         // display extension
         propStr = eglQueryString(mGLDisplay, EGL_EXTENSIONS);
